src/zarr/config.py

```diff
@@ -2,16 +2,52 @@
 
 from typing import Any, Literal, cast
 
-from donfig import Config
+from donfig import Config as DConfig
+
+
+class BadConfigError(ValueError):
+    _msg = "bad Config: %r"
+
+
+class Config(DConfig):  # type: ignore[misc]
+    """Will collect configuration from config files and environment variables
+
+    Example environment variables:
+    Grabs environment variables of the form "DASK_FOO__BAR_BAZ=123" and
+    turns these into config variables of the form ``{"foo": {"bar-baz": 123}}``
+    It transforms the key and value in the following way:
+
+    -  Lower-cases the key text
+    -  Treats ``__`` (double-underscore) as nested access
+    -  Calls ``ast.literal_eval`` on the value
+
+    """
+
+    def reset(self) -> None:
+        self.clear()
+        self.refresh()
+
 
 config = Config(
-    "zarr",
+    "zarr_python",
```

Comment:
I'm curious about the motivation for this change. I think I'd like to keep this as `"zarr"` because that is the name of this python package.

Reply:
Donfig lets you use environment variables to set the configuration starting with the name specified here (e.g. `ZARR_PYTHON_CODEC_PIPELINE__NAME`)
My thoughts were: to avoid confusion about the origin of this environment variable, I wanted to give a hint with "python" 
But I also have no problem with "zarr"

Reply:
I'd stick with `zarr`